File: app/src/main/java/com/andriod/simplenote/NoteApplication.java
package com.andriod.simplenote;

import android.app.Application;
import android.content.Context;

import com.andriod.simplenote.data.BaseDataManager;
import com.andriod.simplenote.data.FirestoreDataManager;
import com.andriod.simplenote.data.PreferencesDataManager;

public class NoteApplication extends Application {

    private static final String SHARED_PREFERENCES_NOTES = "SHARED_PREFERENCES_NOTES";

    private BaseDataManager dataManager;

    @Override
    public void onCreate() {
        super.onCreate();

//        dataManager = new PreferencesDataManager(
//                getApplicationContext()
//                        .getSharedPreferences(SHARED_PREFERENCES_NOTES, Context.MODE_PRIVATE));
        dataManager = new FirestoreDataManager();
    }

    public BaseDataManager getDataManager() {
        return dataManager;
    }
}

File: app/src/main/java/com/andriod/simplenote/NoteApplication.kt
package com.andriod.simplenote

import android.app.Application
import android.content.Context
import com.andriod.simplenote.data.BaseDataManager
import com.andriod.simplenote.data.FirestoreDataManager
import com.andriod.simplenote.data.PreferencesDataManager

class NoteApplication : Application() {
    lateinit var dataManager: BaseDataManager
        private set

    override fun onCreate() {
        super.onCreate()

//        dataManager = PreferencesDataManager(
//            applicationContext
//                .getSharedPreferences(SHARED_PREFERENCES_NOTES, Context.MODE_PRIVATE));
        dataManager = FirestoreDataManager()
    }

    companion object {
        private const val SHARED_PREFERENCES_NOTES = "SHARED_PREFERENCES_NOTES"
    }
}